src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1284,6 +1286,97 @@ private String _checkRenameByField(String implName) {
+    /* Internal methods; merging/fixing case-differences
+    /**********************************************************
+     */
+
+    // @since 2.20
+    protected void _fixLeadingFieldNameCase(Map<String, POJOPropertyBuilder> props)
+    {
+        // 11-Jun-2025, tatu: [databind#5152] May need to "fix" mis-matching leading case
+        //    wrt Fields vs Accessors
+
+        // First: find possible candidates where:
+        //
+        // 1. Property only has Field
+        // 2. Field does NOT have explicit name (renaming)
+        // 3. Implicit name has upper-case for first and/or second character
+
+        Map<String, POJOPropertyBuilder> fieldsToCheck = null;
+        for (Map.Entry<String, POJOPropertyBuilder> entry : props.entrySet()) {
+            POJOPropertyBuilder  prop = entry.getValue();
+
+            // First: (1) and (2)
+            if (!prop.hasFieldAndNothingElse()
+                    || prop.isExplicitlyNamed()) {
+                continue;
+            }
+            // Second: (3)
+            if (!_firstOrSecondCharUpperCase(entry.getKey())) {
+                continue;
+            }
+            if (fieldsToCheck == null) {
+                fieldsToCheck = new HashMap<>();
+            }
+            fieldsToCheck.put(entry.getKey(), prop);
+        }
+        /*// DEBUGGING
+        if (fieldsToCheck == null) {
+            System.err.println("_fixLeadingCase, candidates -> null; props -> "+props.keySet());
+        } else {
+            System.err.println("_fixLeadingCase, candidates -> "+fieldsToCheck);
+        }
+        */
+
+        if (fieldsToCheck == null) {
+            return;
+        }
+
+        for (Map.Entry<String, POJOPropertyBuilder> fieldEntry : fieldsToCheck.entrySet()) {
+            Iterator<Map.Entry<String, POJOPropertyBuilder>> it = props.entrySet().iterator();
+            final POJOPropertyBuilder fieldProp = fieldEntry.getValue();
+            final String fieldName = fieldEntry.getKey();
+
+            while (it.hasNext()) {
+                Map.Entry<String, POJOPropertyBuilder> propEntry = it.next();
+                final POJOPropertyBuilder prop = propEntry.getValue();
+
+                // Skip anything that has Field (can't merge)
+                if (prop == fieldProp || prop.hasField()) {
+                    continue;
+                }
+                if (fieldName.equalsIgnoreCase(propEntry.getKey())) {
+                    // Remove non-Field property; add its accessors to Field one
+                    it.remove();
+                    fieldProp.addAll(prop);
+                    // Should we continue with possible other accessors?
+                    // For now assume only one merge needed/desired
+                    break;
+                }
+            }
+        }
+    }
+
+    // @since 2.20
+    private boolean _firstOrSecondCharUpperCase(String name) {
+         switch (name.length()) {
+         case 0:
+             return false;
+         default:
```

Comment:
Is this neccessary to place default statement in the middle? 

Unless strongly required, Could switch to if else statement

Reply:
Could move `0` as last, making `default` first entry. Not sure if that's any better.

Reply:
Yeah not a blocker. We can go on